hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/execution/traceability/HederaTracer.java

```diff
@@ -203,8 +203,10 @@ private void finalizeActionFor(final SolidityAction action, final MessageFrame f
                     action.setOutput(frame.getOutputData().toArrayUnsafe());
                     if (action.getInvalidSolidityAddress() != null) {
                         // we had a successful lazy create, replace targeted address with its new Hedera id
```

Comment:
Is there a typo in this comment? (At least superficially, it seems if lazy creation was successful, we wouldn't have a non-null `invalidSolidityAddress`?)

Reply:
It looks like the poorly named `invalidSolidityAddress` field is being overloaded to also hold the address in case of a lazy create.  
https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/execution/traceability/HederaTracer.java#L175
https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/execution/traceability/SolidityAction.java#L160
I'm not sure how dangerous it'll be to introduce a new proper field.